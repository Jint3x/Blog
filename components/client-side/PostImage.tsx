import React from "react"

type ImageType = "wide" | "tall";
interface PostImage_I {
    src: string, 
    alt: string,
    desc: string,
    imageType: ImageType
}

function PostImage(props: PostImage_I) {
    return (
        <div className={"pt-4 pb-4"}>
            <img 
             src={props.src}
             alt={props.alt}
             className={`block ml-auto mr-auto ${props.imageType === "wide" ? "w-full sm:w-128" : "w-4/6 sm:w-72 "}`}
            />

            <p className={`ml-auto mr-auto pt-1 text-sm text-color-image-comment ${props.imageType === "wide" ? "w-full sm:w-128" : "w-4/6 sm:w-72 "}`}>
                {props.desc}
            </p>
        </div>
    )
}


export default PostImage