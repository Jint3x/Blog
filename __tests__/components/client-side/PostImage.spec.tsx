/**
 * @jest-environment jsdom
 */
import PostImage from "../../../components/client-side/PostImage";
import ReactRenderer from "react-test-renderer"
import { screen, render } from "@testing-library/react"


test("Renders the component with the correct props", () => {
    render(
        <PostImage 
         src={"/testing/image.png"}
         alt={"A test image"}
         desc={"This is a test text"}
         imageType={"wide"}
        />
    );

    let image = screen.getByAltText("A test image") as HTMLImageElement;
    let text = screen.getByText("This is a test text");

    expect(image.nodeName).toBe("IMG");
    expect(image.src).toBe("http://localhost/testing/image.png");
    expect(text.nodeName).toBe("P");
})


test("Takes a snapshot of the component", () => {
    let tree = ReactRenderer.create(
        <PostImage 
         src={"/testing/image.png"}
         alt={"A test image"}
         desc={"This is a test text"}
         imageType={"wide"}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
})