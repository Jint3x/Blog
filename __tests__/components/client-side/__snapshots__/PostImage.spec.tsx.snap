// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes a snapshot of the component 1`] = `
<div
  className="pt-4 pb-4"
>
  <img
    alt="A test image"
    className="block ml-auto mr-auto w-full sm:w-128"
    src="/testing/image.png"
  />
  <p
    className="ml-auto mr-auto pt-1 text-sm text-color-image-comment w-full sm:w-128"
  >
    This is a test text
  </p>
</div>
`;
